### ISSUE TITLE
Questions, discussions

### ISSUE THREAD

**Issue by ivan-aksamentov** [neherlab/covid19_scenarios#18]
# COVID-19 scenarios: questions and discussions

---

<h1 align="center">
ANNOUNCEMENT
</h1>

<p align="center">
We have a new chat channel, and we'd prefer to move all the conversations there
  <a
    alt="Link to join the chat"
    href="https://spectrum.chat/covid19-scenarios/general/questions-discussions~8d49f461-a890-4beb-84f7-2d6ed0ae503a"
  >
    <img
      alt="Image for the link to join the chat"
      src="https://user-images.githubusercontent.com/9403403/77235704-691ec480-6bb8-11ea-985d-82ec87cfdcdf.png"
    />
  </a>
</p>

---

If you don't want to register there still: 

This thread is dedicated to questions and discussions about the "COVID-19 scenarios" application as well as the science behind it.

### Post here if:

 - you have a question about the scientific aspect of the application
 - you want to tell us about your use-case, to share your experience
 - you want to discuss the potential improvements to the application
 - you have a general suggestion
 - you want to convey some information or criticism
 - you aren't sure whether to create a dedicated [New Issue](https://github.com/neherlab/covid19_scenarios/issues/new/choose) or not

We welcome contributions in any form, as well as suggestions and constructive criticism. We'd be happy to hear about your use-case and workflow. Don't hesitate!

**Comment by rfonseca** [comment]
Hi, what data do we need to get to add a country to the tool? I'm thinking of Brazil, but this is a general question.

**Comment by ivan-aksamentov** [comment]
@rfonseca 
Hi Rodrigo, that would be fantastic!

We are currently trying to make data pipe move convenient for external contributors.

The work is done on branch [`feat/crowdsource-data-submodule`](https://github.com/neherlab/covid19_scenarios/tree/feat/crowdsource-data-submodule)

This include:

  - "crowdsource"- allow people to easily add and modify the data
  - "submodule" - move intermediate data into a separate repository (it's here: [`covid19_scenarios_data`](https://github.com/neherlab/covid19_scenarios_data)) and add it as a submodule of the main repo
  - prepare a guide for data contributors and curators
  - add people willing to curate the data for their chosen country as maintainers to the data repository

More info is coming, but you may take a look on the mentioned branch at:

 - the [`/tools`](https://github.com/neherlab/covid19_scenarios/tree/feat/crowdsource-data-submodule/tools) directory 
 - the [`/data`](https://github.com/neherlab/covid19_scenarios/tree/feat/crowdsource-data-submodule/data) directory 

Basically, scripts download data from all the different sources and transform it into an intermediate format (TSV for example). The intermediate files are committed into the data repository. These files will be then taken on every build, and consumed by the application.

I created an issue for contributor's guide:
 - https://github.com/neherlab/covid19_scenarios/issues/36

Related:
 - https://github.com/neherlab/covid19_scenarios/issues/27

**Comment by kmid5280** [comment]
Hello. I was attempting to explore your application but am having trouble starting it. I followed the installation steps as outlined in the readme. When I run 'yarn dev', I get an error in the console that says 'error Command "dev:nowatch;" not found. Did you mean "dev:nowatch"?' I'm running Windows 10 and installed yarn version 1.22.4 using the downloadable .msi file from the yarn site. Yarn is showing as installed, and typing 'yarn --version' in bash reveals the correct version number. The Windows path variable matches the installation folder. Not sure if this is an issue on my end with the yarn installation itself, or if something is missing in the code. Any tips appreciated!

**Comment by ivan-aksamentov** [comment]
Hi @kmid5280 ,

All of us currently running Linux and there are few Unix-isms still in the config.

I created issue to support Windows: https://github.com/neherlab/covid19_scenarios/issues/38

Meanwhile you may want to take a look at the chain starting from `dev` command in `package.json`. It refers to `nodemon` config file here: `config/nodemon/dev.json`.
Note the `exec` string. Maybe Windows does not like the `;` and `&&` in there?

We would appreciate if you could help!

**Comment by kmid5280** [comment]
@ivan-aksamentov Thanks for the response. I have Docker installed already, so I attempted this method using the above command. It gave me the error 'error An unexpected error occurred: "EIO: i/o error, scandir '/code/node_modules/node-sass/node_modules/glob-parent'".

I thought the same thing you did re: the exec string in the dev.json file, in that it may be a syntax problem. Any suggestions on how to rewrite it? Perhaps there is a way to separate out the 'yarn install' and 'yarn dev:nowatch' sections without using the && operator.

**Comment by ivan-aksamentov** [comment]
@kmid5280 
The docker error is strange. Try to remove `node_modules` `.build` and `.cache` directories and try again.

Without docker, for the time being you can try to run:
```
yarn install

cross-env NODE_ENV=development BABEL_ENV=development babel-node node_modules/.bin/webpack-dev-server --config config/webpack/webpack.client.babel.ts
```

Alternatively, try to run all these commands  in `exec` string separately. `dev:nowatch` is just another script in `package.json` (it runs the second command above).

**Comment by kmid5280** [comment]
@ivan-aksamentov Running 'cross-env NODE_ENV=development BABEL_ENV=development babel-node node_modules/.bin/webpack-dev-server --config config/webpack/webpack.client.babel.ts' produces the error ''babel-node' is not recognized as an internal or external command, operable program or batch file.' Adding the babel-node package via 'yarn add babel-node' doesn't change the error.

Running 'yarn dev:nowatch' produces the error 'node_modules\.bin\webpack-dev-server:2
basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")
          ^^^^^^^
SyntaxError: missing ) after argument list'

If I delete the node_modules and .cache folders (I don't have a .build folder) and then run docker-compose -f docker/docker-compose.dev.yml up --build, then it hangs at the message 'Attaching to covid19_scenarios-dev-web' before timing out.

**Comment by alfredorevilla** [comment]
Hi @ivan-aksamentov, can you provide a couple of samples of Unix-isms that Windows cannot digest as mentioned in #38 ?

Regards.

**Comment by ivan-aksamentov** [comment]
@alfredorevilla I think it may be these semicolumns in the nodemon's `exec`:

https://github.com/neherlab/covid19_scenarios/blob/master/config/nodemon/dev.json#L32

Maybe there's something else. But I cannot investigate this right now.
Also, check the error messages @kmid5280 encountered above.

**Comment by ribrault** [comment]
Hi, 
thanks a lot for this web app, very useful indeed.
From what I got from the severity assumptions table, ca. 90% of severe infections occur in people aged over 60. I'd be very interested in seeing what happens in a scenario where this age class (>60), as a population specifically at risk, would be "confined" for their protection, while the remaining would not be. (such a scenario would certainly bring ethical and practical questions, yet so does a scenario with no age-dependent restrictions). 
I'm sorry I'm not able to code it myself, and I thank you in advance for your answer.
Kind regards,

**Comment by rneher** [comment]
@ribrault This is possible. there is a column `isolation` in the severity table at the bottom left in which you can enter values that specify how much different age groups are separated from the bulk of the population

**Comment by ribrault** [comment]
Indeed. Thanks. I was using a web browser (Safari) where editing of this table was not supported. Adding a warning when parameter editing is not supported by the browser would be helpful.

On the simulations: it looks like confining the population at risk (e.g. > 60 yo) while partly releasing confinement of the rest of the population allows to avoid ICU overflow and to importantly reduce the total number of deaths… do you know whether this strategy has been considered in practice and do you have any opinion on that ? 

Also, please tell me which channel is most appropriate to discuss that, if not here…

**Comment by darwinbandoy** [comment]
My computed reproductive number is 2.5 for the Philippines with temperatures around 28 degree celsius. I don't this will share the seasonality of other diseases. The default seasonality I suggest should be zero.

**Comment by DanielZug** [comment]
@ribault, I had very similar thoughts and it's great to have this simulation tool available to play with! I'm quite unhappy about the prospect of living / working with the restrictions in place for many months to come and desperately look for an exit strategy. (Of course, in the current situation, we must at first get the outbreak under control before anything else.)
One important question for me is how to ensure that the virus is not transmitted by health care staff that is taking care of the elderly people. Actually one would not only have to isolate the >60 year olds and other persons at risk from the rest of the population, but also their regular contacts, unless the risk of transmissions can be safely reduced.
And then, of course, it's very important to validate the assumptions or parameters of the model with reality before starting a large-scale experiment!
In my view, we need to learn to live _with_ the virus, like any other, we won't be able to eliminate it worldwide. So we need to ban the risk of an epidemic that threatens large parts of our population every few months. It would be interesting to hear what scenarios scientists and polticians are considering for this and on what basis decisions will be made.

**Comment by jc92** [comment]
Apologies for opening a ticket, just discovered this thread.

I'm trying to find granular data on confirmed cases, but haven't been successful to do so until now. Could you please share how you are getting the numbers from BAG?
Thanks in advance.

**Comment by richardhodgman** [comment]
Can mitigation end time be extended?  I would like to extend it like simulation time can be extended.  My test cases are breaking down because mitigation ends September 30.  
Making mitigation time equal to as simulation time would be OK.

**Comment by vejmelkam** [comment]
Hi guys, great work! I have a question related to modeling the effect of healthcare capacity.

The model parameters ask for all beds/ICUs available in the country.

- How is the proportion of beds/ICUs available to tread COVID-19 patients computed?
- Is there differentiation between recovery rate of patients that are hospitalized vs patients that are in the "overflow" group?

**Comment by nnoll** [comment]
@vejmelkam 

As of now, we assume that the full number of ICU beds are available to treat COVID-19 patients. If one wanted to model a smaller number, the easiest way to do this right now is to decrease the number of ICU beds in the simulation parameters.

It is important to note that the patients who end up in the ICU don't natively recover in this model, they transition back to the hospital state. The rate at which ICU patients and overflow patients transition back to the hospital are equivalent, however the rate of death of overflow patients is higher (and tunable by the overflow severity parameter).

**Comment by nnoll** [comment]
@richardhodgman 

The mitigation time should automatically update when you update the simulation window. Does it not for you? If so can you send a screenshot?

**Comment by vejmelkam** [comment]
@nnoll thank you for your response.

Understood about hospital capacity. Would it make sense to add this to the description of the hospital/ICU beds parameter as a warning? For example in the Czech Republic, the numbers are a huge difference. Only a tiny fraction of the hospital bed capacity is expected to be available (thousands, as far as imprecise specs in news articles go). I'm happy to send a PR if interested.

Regarding the modelling assumption of transition rates
- critical->severe
- severe->recovered
being independent of whether hospital capacity is available for those patients, would you consider changes to the model here? This would require the introduction of new parameters for the overflow patients in each compartment and corresponding changes to the equations. I would be happy to explore this and help with some variants here.

We could start with only one of these links, e.g. severe->recovered transition. Would you be at all interested in such a contribution?

**Comment by nnoll** [comment]
@vejmelkam thank you for your interest and help!

A disclaimer is definately warranted. If you feel up for it, I would hapily accept a PR with this change. 

I'm not sure I fully understand your proposed changes to the model. Are you thinking of introducing another overflow class to the hospital that explictly models finite hospital beds, in addition to the imposed ICU constraints? If so, what would your assumptions be about the flux into and out of this state? 

In general, we're trying to keep the number of tunable knobs in a manageable regime, however I am always interested in making the model more realistic. If you feel inclined, try to explore your idea and we can look at the results and think about integrating into master.

**Comment by vejmelkam** [comment]
@nnoll 

Essentially this is my current thinking. Flow assumptions:

- infected-> severe is unchanged but hospital bed assignment follow similar logic as ICU in the [model.js](https://github.com/neherlab/covid19_scenarios/blob/0a73b6ed7bf408cf1189b24cf09f8296dbd3263a/src/algorithms/model.js#L218) file.

- severe->critical is split into two flows. The new flow is for severe AND non-hospitalized people that would have a higher chance for going into critical state (add single extra multiplier just like ICU severity). The macro effect here is to model increased pressure on ICUs if hospital beds are unavailable.

The reason I think this effect may turn out to be a significant effect is because the actual hospital capacity will be grossly insufficient (compared to the current pre-sets which indicate total capacity).

I am happy to do the change locally and investigate the effect before sending a PR. However I'd do this only if there is a chance of upstream acceptance.

**Comment by richardhodgman** [comment]
Screen shot attached.  Running Chrome Version 80.0.3987.149 (Official
Build) (64-bit) on Windows 10 Home version 1909 OS Build 18363.720 up to
date 2020-03-18
============================
Richard Hodgman
richard@hodgman.org
http://hodgman.org/

**Comment by nnoll** [comment]
@vejmelkam 

This sounds great. I thought about a similar addition but decided against just for simplicity. I also convinced myself it wouldn't make a qualitative difference. However, if you implement it and show it makes significant changes to the predictions at the expense of one/two parameters added, I would say this is a meaningful change to merge upstream. I would say play around and report back!

**Comment by VascoVisser** [comment]
I noticed that increasing the infectious period decreases the severity of the simulated epidemic. Is that a bug, or does it make sense somehow?

**Comment by wscott** [comment]
@VascoVisser just guessing here, but if you don't change R0 (number of infections per case), but you do increase the infection periods then effectively you are slowing down the spread and giving more time for hospitals and mitigations to work.  In reality, if the infections period was longer then R0 would also go up.

**Comment by VascoVisser** [comment]
That makes sense! Thanks.

**Comment by lukepighetti** [comment]
Can we get a high level overview of what this project's goal is. README says it's to model hospital demand, but I also see reference to flow-chart style "what to do next" scenario guidelines. I've also seen reference to current stats on the outbreak. Some clarity would be appreciated, even if that is that there is no explicit direction.

**Comment by pnwscm60** [comment]
I did a very simple model just to cover the first 31 days of the outbreak, just to show, how important the measures taken really are (https://ebpswiss.ch/animation.html). Using your tool (nice work!), I have lower counts of deaths and ICU need in the beginning. Can you give me your basic assumptions about deaths and ICU need? Thanks!

**Comment by PaulKC** [comment]
I really like the tool. It would be nice if you could also simulate the seasonal flu to have a comparison.

**Comment by visionscaper** [comment]
Hello,

**Comment by visionscaper** [comment]
Hello,

Great simulation tool, thank you for sharing!

I have a question : does your model take in to account the effects where the rate of new infections slow due to the herd immunity effect?

**Comment by bharath6365** [comment]
I have spotted a few areas where usability/user-experience could be improved. @ivan-aksamentov  
Can we sync up?

**Comment by pinanunes** [comment]
I've got one doubt and one suggestion:

Doubt: When modeling critical cases vs existing ICU beds are you accounting for usual ICU bed occupancy? It is not a big issue since we can enter the number of ICU beds like the ones available taking into account the available beds. But nevertheless is good to consider that.

Suggestion: One major question at the moment is the effect of testing to increase the identification and consequent quarantine of positives. Most european countries are following ECDC's suspect definition (https://www.ecdc.europa.eu/en/case-definition-and-european-surveillance-human-infection-novel-coronavirus-2019-ncov) that test mostly severe cases. One option for the tool would be to have a model for mitigation that would include testing coverage effect on baseline mitigation measures. If you want to explore this I'm up for contribution/discussions

**Comment by nnoll** [comment]
@visionscaper Yes, this model captures "herd-immunity" as we assume recovered individuals can not become re-infected. This is easiest to see in a simple S(Susceptible)-I(Infected)-R(Recovered) model. I'll speak in ODE's, let me know if I should provide more details :)

The number of infected[I] grows in time as (N denotes the population size):

dI/dt = b*S*I/N -  c*I

At steady state (when infections don't grow or shrink in time) dI/dt = 0 , or when S/N = c/b. c/b is the "herd immunity" fraction. Hope this helps

**Comment by nnoll** [comment]
@pinanunes So far there is no modelling of occupancy of ICU beds: we assume all  beds can be used by COVID-19 patients. There is a disclaimer in the help that states this assumption, although I agree it could be more prominent. Do you have any suggestions?

Regarding your suggestion, this sounds interesting although I'm not sure exactly what you have in mind. Would you mind outlining your thoughts a bit more on this?

**Comment by nnoll** [comment]
@pnwscm60 All our data that we use regarding hospital and ICU utilization can be found within the element labelled "Severity assumptions". Also, these assumptions can be edited if you want to use your own data to match your model.

**Comment by wiedenhoeft** [comment]
Hi, this is great work!! I was wondering, given the potentially large number of subsymptomatic, undetected cases, whether "I" should be split up into a symptomatic and an unsymptomatic compartment, or diagnosed vs. undiagnosed. Given the wide range of symptoms, the infected are potentially underreported, and the proportion of those cases might become an important model parameter should large-scale test testing of random subpopulations ever be available.

**Comment by nnoll** [comment]
@lukepighetti yes we are putting together a more detailed roadmap. should be together shortly.

Also, for everyone but also regarding community organization: I welcome you to head over to https://spectrum.chat/covid19-scenarios to continue discussion.  We think this should facilitate more productive community discussion and collaboration.

We'll leave this thread open for awhile until everyone can migrate. Thanks again for all your suggestions and help!

**Comment by chriswien** [comment]
Hi, 
I fetched covid19_scenarios_data  and covid19_scenarios to the same folder. 
I did some work on regional datafor germany and I wrote the parser. I was about testing if works.
So I followed your guidelines from covid19_scenarios_data module and tried this:

python3 parsers/germany.py

Traceback (most recent call last):
  File "parsers/germany.py", line 7, in <module>
    from .utils import write_tsv
ModuleNotFoundError: No module named '__main__.utils'; '__main__' is not a package

What's the trick? Sorry I am not a python guy

Greetings
Christian

**Comment by nnoll** [comment]
@chriswien see https://stackoverflow.com/questions/14132789/relative-imports-for-the-billionth-time for a long discussion. The short answer is .(python file name) is a relative path import that is only expected to work when you use your parser from the parsers module. 

One easy trick is to call python parse_all.py while in the data root directory, which should take care of this particular issue for you. Check out ecdc.py as an example if you would like to run this as a standalone script for debugging.

**Comment by chriswien** [comment]
thanks that works ,but now an error occurs running ecdc.py
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 1317, in do_open
    encode_chunked=req.has_header('Transfer-encoding'))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1229, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1275, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1224, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1016, in _send_output
    self.send(msg)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 956, in send
    self.connect()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1392, in connect
    server_hostname=server_hostname)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ssl.py", line 412, in wrap_socket
    session=session
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ssl.py", line 853, in _create
    self.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ssl.py", line 1117, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1056)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "parse_all.py", line 12, in <module>
    country.parse()
  File "/Users/christian/Desktop/Covid19_scenarios/covid19_scenarios_data/parsers/ecdc.py", line 111, in parse
    cases = retrieve_case_data()
  File "/Users/christian/Desktop/Covid19_scenarios/covid19_scenarios_data/parsers/ecdc.py", line 55, in retrieve_case_data
    file_name, headers = urlretrieve(URL+date+".xlsx")
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 247, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 543, in _open
    '_open', req)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 1360, in https_open
    context=self._context, check_hostname=self._check_hostname)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py", line 1319, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1056)>

**Comment by chriswien** [comment]
Hi,
I like to provide data for germany.

I pull data from Robert Koch Institut RKI on a daily base. The have actual data 10:00 each day which reflect the state 0:00 the same day. This is the link

https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Fallzahlen.html

I have created a git repository with a csv file for data beginning on Feb 24 2020. (source: https://de.wikipedia.org/wiki/COVID-19-Pandemie_in_Deutschland and https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Fallzahlen.html)

https://github.com/chriswien/COVID19_CaseNumberDE

Also available is a python file to strip the data table and an sed file for editing if you like.

Cheers
Christian 


Christian Wienbruch
Brauneggerstraße 32
78462 Konstanz

**Comment by DrGo** [comment]
Hello,
Great app!
Would it be possible to run the simulation code from the command line (using node)?

**Comment by ivan-aksamentov** [comment]
Hello, thanks everyone for the participation!

We've created a Spectrum chat, where communication should be somewhat easier:

<p align="center">
  <a
    alt="Link to join the chat"
    href="https://spectrum.chat/covid19-scenarios/general/questions-discussions~8d49f461-a890-4beb-84f7-2d6ed0ae503a"
  >
    <img
      alt="Image for the link to join the chat"
      src="https://user-images.githubusercontent.com/9403403/77235704-691ec480-6bb8-11ea-985d-82ec87cfdcdf.png"
    />
  </a>
</p>

We will eventually address all the questions and ideas here too, of course.

**Comment by ivan-aksamentov** [comment]
@DrGo 
Not right now. An it's not in our plans currently. But that should be fairly easy to implement. Algorithm is contained in `src/algorithms` and the function `run()`, the entry point, is in `run.ts`.

Feel free to open an issue, describe the user experience you imagine, and maybe someone will implement it. Or feel free to try yourself and then open a pull request. If it flies, we can organize an npm package too.

**Comment by ivan-aksamentov** [comment]
cc @nnoll @rneher (Data from Germany)

@chriswien Thanks! 

I opened an issue here:
https://github.com/neherlab/covid19_scenarios_data/issues/21

If you are available to help us to preprocess this data, feel free to check our data sourcing thread here:
https://github.com/neherlab/covid19_scenarios/issues/69
and the data contributor's guide here:
https://github.com/neherlab/covid19_scenarios_data

For any questions don't hesitate to post in the "Data sources" chat
https://spectrum.chat/covid19-scenarios/data-sources/discussion-forum-for-suitable-sources-of-covid19-case-data~350022e0-260d-420a-9387-6cf4c8428308

**Comment by rrickfox** [comment]
Hi,
I tried to implement the model you describe on your website [https://neherlab.org/covid19/about](url). However I got stuck on understanding the differential equation for I_a, in particular what the parameter I_b stands for. I was unable to find the equivalent in the code of this repo. In my case the infectionchange increases rapidly, as the formula I use is dI_a/dt = -beta * S_a * I_a. I am certain that this formula is wrong.
Any help would be appreciated.
Best regards, Richard

**Comment by ivan-aksamentov** [comment]
A link to chat for the issue raised by @rrickfox 
https://spectrum.chat/covid19-scenarios/general/questions-discussions~8d49f461-a890-4beb-84f7-2d6ed0ae503a?m=MTU4NDgzNDMzMjc1OQ==

**Comment by GD-ZH** [comment]
Hi!
Great easy-to-use model you have createt! I have two questions:
1. The colour of the "confirmed cases"-dots and the "infectious"-line match like the "deaths" dots and lines. However in my understanding they should not fit because not all the infectious cases are reported. Is this understanding correct?
2. when i download a csv file, I get a file without commas. Is this possibly a bug?
Keep up the great work!
Stefan

**Comment by ivan-aksamentov** [comment]
@GD-ZH  Hi Stefan, 

1. I will let our scientists @nnoll and @rneher to address that

2. Yes, sort of. It is a `.tsv` (tab-separated values) file but named as `.csv`. We switched some time ago but forgot to rename. I will fix that.

P.S. Feel free to also join our chat at this link:
https://spectrum.chat/covid19-scenarios

**Comment by reitzig** [comment]
I miss _any_ indication of confidence. Either in the statistical sense, or something along the lines of applying the model with a similar "prefix" of the date to a past pandemic, or even China during this one. If the model can't predict known data, I'm less inclined to believe in it's predicions.

**Comment by nnoll** [comment]
This is because, in it's current form, the model is fully deterministic: you give it a set of input parameters and it will produce an estimate for the expected number of individuals in each class over time. You are right that this has no uncertainty displayed currently.

You should note that our model is not fit to case counts yet (this should be finished soon). You can certainly fiddle with parameters to get predictive curves -- there are more than enough parameters to this model to do so.

I think it's important for all of us to discuss uncertainty as there are two potential sources here. One, the underlying dynamics are stochastic which is not captured by our simple system of ODE's. We believe this is subdominant to a larger source of uncertainty -- our inability to accurately (quantitatively) know the epidemiological parameters. And here I completely agree with you, our model could do better. And it's something we are actively working on, users should be able to specify error bars to all parameters and the output should give them a distribution of outcomes that sample from this parameter space. We are just starting to work on this and would welcome contributions in this direction.
